diff --git a/src/test/clojure/clojure/core/match/test/core.clj b/src/test/clojure/clojure/core/match/test/core.clj
index b85d88d..1405938 100644
--- a/src/test/clojure/clojure/core/match/test/core.clj
+++ b/src/test/clojure/clojure/core/match/test/core.clj
@@ -526,9 +526,9 @@
 (deftest vector-pattern-match-1
   (is (= (let [x [1 2 3]]
            (match [x]
-             [([_ _ 2] ::clojure.core.match/vector)] :a0
-             [([1 1 3] ::clojure.core.match/vector)] :a1
-             [([1 2 3] ::clojure.core.match/vector)] :a2
+             [([_ _ 2] :clojure.core.match/vector)] :a0
+             [([1 1 3] :clojure.core.match/vector)] :a1
+             [([1 2 3] :clojure.core.match/vector)] :a2
              :else :a3))
         :a2)))
 
@@ -595,9 +595,9 @@
 (deftest vector-pattern-rest-1
   (is (= (let [v [1 2 3 4]]
            (match [v]
-             [([1 1 3 & r] ::clojure.core.match/vector)] :a0
-             [([1 2 4 & r] ::clojure.core.match/vector)] :a1
-             [([1 2 3 & r] ::clojure.core.match/vector)] :a2
+             [([1 1 3 & r] :clojure.core.match/vector)] :a0
+             [([1 2 4 & r] :clojure.core.match/vector)] :a1
+             [([1 2 3 & r] :clojure.core.match/vector)] :a2
              :else :a3))
         :a2)))
 
@@ -605,8 +605,8 @@
   (is (= (let [v [1 2 3 4]]
            (let [v [1 2 3 4]]
              (match [v]
-               [([1 1 3 & r] ::clojure.core.match/vector)] :a0
-               [([1 2 & r] ::clojure.core.match/vector)] :a1
+               [([1 1 3 & r] :clojure.core.match/vector)] :a0
+               [([1 2 & r] :clojure.core.match/vector)] :a1
                :else :a3)))
         :a1)))
 
