diff --git a/src/main/clojure/clojure/tools/namespace/dependency.clj b/src/main/clojure/clojure/tools/namespace/dependency.clj
index 61c75ff..a1a8c3d 100644
--- a/src/main/clojure/clojure/tools/namespace/dependency.clj
+++ b/src/main/clojure/clojure/tools/namespace/dependency.clj
@@ -146,8 +146,7 @@
                (clojure.set/union (set more) (set add)))))))
 
 (def ^:private max-number
-  #?(:clj Long/MAX_VALUE
-     :cljs js/Number.MAX_VALUE))
+  Long/MAX_VALUE)
 
 (defn topo-comparator
   "Returns a comparator fn which produces a topological sort based on
diff --git a/src/main/clojure/clojure/tools/namespace/dir.clj b/src/main/clojure/clojure/tools/namespace/dir.clj
index e12288d..7280f85 100644
--- a/src/main/clojure/clojure/tools/namespace/dir.clj
+++ b/src/main/clojure/clojure/tools/namespace/dir.clj
@@ -20,10 +20,15 @@
   (:import (java.io File) (java.util.regex Pattern)))
 
 (defn- find-files [dirs platform]
-  (->> dirs
-       (map io/file)
-       (filter #(.exists ^File %))
-       (mapcat #(find/find-sources-in-dir % platform))))
+  (let [all-files (->> dirs
+                       (map io/file)
+                       (filter #(.exists ^File %))
+                       (mapcat file-seq)
+                       (remove #(.isDirectory ^File %)))
+        {:keys [extensions]} (or platform find/clj)
+        grouped (group-by #(file/file-with-extension? % extensions) all-files)]
+    {:clojure-files (get grouped true)
+     :non-clojure-files (get grouped nil)}))
 
 (defn- modified-files [tracker files]
   (filter #(< (::time tracker 0) (.lastModified ^File %)) files))
@@ -86,8 +91,10 @@
   ([tracker] (scan-dirs tracker nil nil))
   ([tracker dirs] (scan-dirs tracker dirs nil))
   ([tracker dirs {:keys [platform add-all?] :as options}]
-   (let [ds (or (seq dirs) (classpath-directories))]
-     (scan-files tracker (find-files ds platform) options))))
+   (let [ds (or (seq dirs) (classpath-directories))
+         {:keys [clojure-files non-clojure-files]} (find-files ds platform)]
+     (assoc (scan-files tracker clojure-files options)
+            ::non-clojure-files non-clojure-files))))
 
 (defn scan
   "DEPRECATED: replaced by scan-dirs.
diff --git a/src/main/clojure/clojure/tools/namespace/parse.clj b/src/main/clojure/clojure/tools/namespace/parse.clj
index 0699c94..7fa4472 100644
--- a/src/main/clojure/clojure/tools/namespace/parse.clj
+++ b/src/main/clojure/clojure/tools/namespace/parse.clj
@@ -10,8 +10,7 @@
       :doc "Parse Clojure namespace (ns) declarations and extract
   dependencies."}
   clojure.tools.namespace.parse
-  (:require #?(:clj [clojure.tools.reader :as reader]
-               :cljs [cljs.tools.reader :as reader])
+  (:require [clojure.tools.reader :as reader]
             [clojure.set :as set]))
 
 (defn comment?
